### PR TITLE
MM-54814: Convert ./components/widgets/menu/menu_group.tsx from Class Component to Function Component

### PR DESCRIPTION
**Pull request by KuSh**
#### Summary

Convert ./components/widgets/menu/menu_group.tsx from Class Component to Function Component

#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/24775
Jira https://mattermost.atlassian.net/browse/MM-54814

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/widgets/menu/menu_group.tsx [hunk 1]
```diff
@@ -8,32 +8,30 @@ import './menu_group.scss';
...
 
-        return (
-            <React.Fragment>
-                {divider}
-                {children}
-            </React.Fragment>
-        );
-    }
-}
+export default MenuGroup;
```
Comment (larkox) [inline webapp/channels/src/components/widgets/menu/menu_group.tsx:37]: The component used to be a pure component, so add here `React.memo`.

**Review by larkox — CHANGES_REQUESTED** [review]
Great work! Just one small thing.

File: webapp/channels/src/components/widgets/menu/menu_group.tsx (hunk 1 above)
Comment (KuSh) [reply webapp/channels/src/components/widgets/menu/menu_group.tsx]: Oh nice trick ! Sure done

**Review by sinansonmez — APPROVED** [review]
LGTM, thanks for the effort

**Comment by larkox** [comment]
@KuSh There are failing tests. Most of them (if not all) can be fixed by running `npm run test -- -u` in the channels folder. Would you mind fixing them?

**Comment by KuSh** [comment]
> @KuSh There are failing tests. Most of them (if not all) can be fixed by running `npm run test -- -u` in the channels folder. Would you mind fixing them?

Sure !

**Comment by KuSh** [comment]
I've fixed tests and updated snapshots @larkox

**Comment by larkox** [comment]
Sorry for the back and forth @KuSh . There seems to be now some conflict with master. Can you merge master into your branch and fix the conflict?

**Comment by KuSh** [comment]
> Sorry for the back and forth @KuSh . There seems to be now some conflict with master. Can you merge master into your branch and fix the conflict?

No problem! I've rebased and fixed the conflict @larkox

**Comment by larkox** [comment]
@KuSh There seems to be another merge conflict. Really sorry for that :sweat:

**Comment by KuSh** [comment]
Done @larkox 👍

**Review by larkox — APPROVED** [review]
Thank you very much!

**Review by lindy65 — APPROVED** [review]
Thanks @KuSh looks good to me 👍 

cc @larkox

**Pull request merged**